### PR TITLE
Singapore holidays update

### PR DESCRIPTION
**Pull request by KJhellico**
- observed holidays calculation fixed
- some holidays dates fixed

### CODE REVIEW

File: holidays/countries/singapore.py [hunk 1]
```diff
@@ -84,6 +84,10 @@ def __init__(
         super().__init__(years, expand, observed, subdiv, prov, state)
 
     def _populate(self, year) -> None:
+        def _add_holiday(dt: date, hol: str) -> None:
```
Comment (arkid15r) [inline holidays/countries/singapore.py:87]: The name could be just `add_holiday(dt: date, hol: str) -> None:` as that's an inner function with no name conflicts.

**Review by arkid15r — COMMENTED** [review]
🇸🇬 LGTM!

File: holidays/countries/singapore.py (hunk 1 above)
Comment (KJhellico) [reply holidays/countries/singapore.py:87]: Many countries with islamic holidays have this function exactly as `_add_holiday`, so I think it's better to be uniformly for now.

Comment (dr-prodigy) [reply holidays/countries/singapore.py:87]: Agree w @KJhellico , the underscore suggests it's a private member 👍

**Comment by arkid15r** [comment]
Great work on code and test improvements!

**Pull request merged**